DocumentsFromSnapshotMigration/src/main/java/com/rfs/RfsMigrateDocuments.java

```diff
@@ -182,63 +185,64 @@ public static void main(String[] args) throws Exception {
-                DocumentReindexer reindexer = new DocumentReindexer(targetClient,
-                    arguments.numDocsPerBulkRequest,
-                    arguments.numBytesPerBulkRequest,
-                    arguments.maxConnections);
-
-                SourceRepo sourceRepo;
-                if (snapshotLocalDirPath == null) {
-                    sourceRepo = S3Repo.create(
-                        Paths.get(arguments.s3LocalDir),
-                        new S3Uri(arguments.s3RepoUri),
-                        arguments.s3Region
-                    );
-                } else {
-                    sourceRepo = new FileSystemRepo(snapshotLocalDirPath);
-                }
-                DefaultSourceRepoAccessor repoAccessor = new DefaultSourceRepoAccessor(sourceRepo);
-
-                var sourceResourceProvider = ClusterProviderRegistry.getSnapshotReader(arguments.sourceVersion, sourceRepo);
-
-                SnapshotShardUnpacker.Factory unpackerFactory = new SnapshotShardUnpacker.Factory(
-                    repoAccessor,
-                    luceneDirPath,
-                    sourceResourceProvider.getBufferSizeInBytes()
+            OpenSearchClient targetClient = new OpenSearchClient(connectionContext);
+            DocumentReindexer reindexer = new DocumentReindexer(targetClient,
+                arguments.numDocsPerBulkRequest,
+                arguments.numBytesPerBulkRequest,
+                arguments.maxConnections);
+
+            SourceRepo sourceRepo;
+            if (snapshotLocalDirPath == null) {
+                sourceRepo = S3Repo.create(
+                    Paths.get(arguments.s3LocalDir),
+                    new S3Uri(arguments.s3RepoUri),
+                    arguments.s3Region
                 );
+            } else {
+                sourceRepo = new FileSystemRepo(snapshotLocalDirPath);
+            }
+            var repoAccessor = new DefaultSourceRepoAccessor(sourceRepo);
 
-                run(
-                    LuceneDocumentsReader.getFactory(sourceResourceProvider),
-                    reindexer,
-                    workCoordinator,
-                    arguments.initialLeaseDuration,
-                    processManager,
-                    sourceResourceProvider.getIndexMetadata(),
-                    arguments.snapshotName,
-                    arguments.indexAllowlist,
-                    sourceResourceProvider.getShardMetadata(),
-                    unpackerFactory,
-                    arguments.maxShardSizeBytes,
-                    rootDocumentContext
-                );
-            });
+            var sourceResourceProvider = ClusterProviderRegistry.getSnapshotReader(arguments.sourceVersion, sourceRepo);
+
+            var unpackerFactory = new SnapshotShardUnpacker.Factory(
+                repoAccessor,
+                luceneDirPath,
+                sourceResourceProvider.getBufferSizeInBytes()
+            );
+
+            run(RunParameters.builder()
```

Comment:
Initially I wasn't sold on fixing `Method has 12 parameters, which is greater than 7 authorized.`, but I figured I'd make a builder from a pojo for the arguments and see how it turned out.

I think this is actually much easier to read and reorder without causing a breaking changes, a decent improvement for a small effort.